account_invoice_change_currency/models/account_change_currency.py

```diff
@@ -130,14 +136,19 @@ def get_last_currency_id(self, skip_update_currency=False):
         query = """
 SELECT mtv.old_value_integer, mtv.new_value_integer, mm.subtype_id
 FROM mail_tracking_value as mtv INNER JOIN mail_message AS mm
-ON mtv.mail_message_id = mm.id """
+ON mtv.mail_message_id = mm.id INNER JOIN
+ir_model_fields AS field
+ON field.id = mtv.field """
         if skip_update_currency:
             query += " AND mm.subtype_id != %s "
-            params = (subtype_id.id, tuple(self.message_ids.ids))
+            params = (subtype_id.id, self._name, tuple(self.message_ids.ids))
         else:
```

Comment:
Hi @rolandojduartem This `else` can be gone if the `params` are set before the `if skip_update_currency`

Reply:
Done!